app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.java

```diff
@@ -85,11 +89,13 @@ public void onPause() {
             activeAsyncTask = null;
             menuItemForActiveAsyncTask = null;
         }
+        imm.hideSoftInputFromWindow(getView().getWindowToken(),0);
     }
 
     @Override
     public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
         inflater.inflate(R.menu.menu_search_engine_manual_add, menu);
+        imm.toggleSoftInput(InputMethodManager.SHOW_FORCED,InputMethodManager.HIDE_IMPLICIT_ONLY);
```

Comment:
Why is toggle necessary? Also, why are the SHOW_FORCED & HIDE_IMPLICIT flags necessary? WHen can a user force the soft input open?

Reply:
Because showSoftInput doesn't work here. I think the reason is the Activity is not "focused" so to use showSoftInput, i first need to call setFocusable and setFocusableInTouchMode. But if i toggle, it works just like that.

Reply:
Maybe this is related to why ViewUtils isn't working...

What else is focused? Would it be appropriate to allow focus on the activity? I'm surprised anything else besides the activity can be focused. Note: you can use the [UI hierarchy tool](https://developer.android.com/studio/profile/hierarchy-viewer.html) to figure out what is focused.